providers/ofrep/src/main/java/dev/openfeature/contrib/providers/ofrep/internal/OfrepApi.java

```diff
@@ -0,0 +1,127 @@
+package dev.openfeature.contrib.providers.ofrep.internal;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import java.io.IOException;
+import java.net.ProxySelector;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.Executor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * The OfrepApi class is responsible for communicating with the OFREP server to evaluate flags.
+ */
+@Slf4j
+public class OfrepApi {
+
+    private static final String path = "/ofrep/v1/evaluate/flags/";
+    private final ObjectMapper serializer;
+    private final ObjectMapper deserializer;
+    private final HttpClient httpClient;
+    private Instant nextAllowedRequestTime = Instant.now();
+
+    /**
+     * Constructs an OfrepApi instance with a HTTP client and JSON serializers.
+     *
+     * @param timeout        - The timeout duration for establishing HTTP connection.
+     * @param proxySelector  - The ProxySelector to use for HTTP requests.
+     * @param executor       - The Executor to use for operations.
+     */
+    public OfrepApi(Duration timeout, ProxySelector proxySelector, Executor executor) {
+        httpClient = HttpClient.newBuilder()
+                .connectTimeout(timeout)
+                .proxy(proxySelector)
+                .executor(executor)
+                .build();
+        serializer = new ObjectMapper();
+        deserializer = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    /**
+     * prepareHttpRequest is preparing the request to be sent to the OFREP Server.
+     *
+     * @param url         - url of the request
+     * @param headers     - headers to be included in the request
+     * @param requestBody - body of the request
+     *
+     * @return HttpRequest ready to be sent
+     * @throws JsonProcessingException - if an error occurred while processing the json.
+     */
+    private <T> HttpRequest prepareHttpRequest(
+            final URI url, ImmutableMap<String, ImmutableList<String>> headers, final T requestBody)
+            throws JsonProcessingException {
+
+        HttpRequest.Builder reqBuilder = HttpRequest.newBuilder()
+                .uri(url)
+                .header("Content-Type", "application/json; charset=utf-8")
```

Comment:
can leave _requestTimeout_ here and have both _requestTimeout_ and _connectTimeout_ to be configurable.